OpenSim/Common/ComponentList.h

```diff
@@ -168,6 +168,10 @@ class ComponentListIterator :
     bool operator==(const ComponentListIterator& iter) const {
         return _node == &*iter;
     }
+    /// Check for equality using method rather than operator to support scripting
+    bool equals(const ComponentListIterator& iter) const {
```

Comment:
Shouldn't this method have the exact same implementation as `operator==`? It seems very different -- I think one is comparing node addresses and the other is comparing iterators. I would like to see just one implementation (the right one, whichever that is!), for `equals()`, and then `operator==` should just invoke that implementation.